notebooks/ch14/bias_modeling.md

```diff
@@ -216,14 +252,22 @@ plt.title('Irreducible error');
+
+Cross-validation provides a more accurate method of estimating our model error using a single observed dataset by separating data used for training from the data used for model selection and final accuracy. Intuitively, the validation or test error checks the model's performance on a previously unseen dataset and thus allows us to estimate both bias and variance. To choose the final model to use, we select the one that has the lowest validation error.
+
+The simplest form of cross-validation is the **holdout method**. This involves splitting the dataset into a training set and validation set. The model is fit on the training set and then is used to predict the unseen values in the validation set. This allows us to judge the generalization error by using an independent sample of data, measuring both the model bias and variance. This method is better than just having a training/test split, and doesn't take any overhead to compute. However, the holdout method is prone to high variance, because the evaluation of the error may depend heavily on which points end up in the training and validation sets. 
+
+To tackle this problem, we can run the holdout method multiple times on the same dataset. The dataset is divided into *k* subsets, and the holdout method is repeated k times. Each time, one of the *k* subsets is used as the validation set, and the other *k-1* subsets are combined and used as the training set. The validation error is computed as the average error from all *k* trials. This method is called **k-fold cross-validation**. The biggest advantage of this method is that every data point is used for validation exactly once, and for training *k-1* times. Typically, a *k* between 5 to 10 is used, but *k* remains an unfixed parameter. The smaller *k* is, the fewer computations one has to do. Also, the error estimate has a lower variance (many validation points), but has a higher bias (fewer training points). Vice versa, with large *k*, there are more computations, and the error estimate will have lower bias, but have higher variance. K-fold cross-validation outputs a prediction error that more accurately reflects the bias and variance of a model. The model bias is the validation error. If this value is low, this means that your model predicts values with low error on average. The model variance can be estimated by taking the standard deviation of the errors in each of the *k* trials. If the value is high, it means that the model's performance varies a lot with the dataset used for training. 
+
```

Comment:
For this entire section, I think we should talk to Sam and find a way to combine the current 13.3 with what you have here. There is a lot of repetition, but some of the points you mention about bias-variance are important too. I also feel it is not necessary to mention the "holdout method" since it is pretty similar to normal train-test split, and we never really mention the term in lecture.

Reply:
talked to sam about this